src/Castle.Core/DynamicProxy/ProxyGenerator.cs

```diff
@@ -918,7 +914,7 @@ public TClass CreateClassProxyWithTarget<TClass>(TClass target, params IIntercep
 		///   This method uses <see cref = "IProxyBuilder" /> implementation to generate a proxy type.
 		///   As such caller should expect any type of exception that given <see cref = "IProxyBuilder" /> implementation may throw.
 		/// </remarks>
-		public object CreateClassProxyWithTarget(Type classToProxy, Type[] additionalInterfacesToProxy, object target,
+		public object CreateClassProxyWithTarget(Type classToProxy, Type[]? additionalInterfacesToProxy, object target,
```

Comment:
I didn't realise until just now that `CreateClassProxyWithTarget` accepts `null` values for `target` (plus it allows the target to be modified via `IProxyTargetAccessor`), though I am not sure whether that is an error or by design?

Reply:
Annotated `target` too as `object?`. (At least for the time being. If this shouldn't actually be allowed, it could be fixed in a separate PR.)